build-support/functions/10-util.sh

```diff
@@ -974,7 +974,7 @@ function ui_version {
       return 1
    fi
 
-   local ui_version="$(grep '<!-- CONSUL_VERSION: .* -->$' "$1" | sed 's/^<!-- CONSUL_VERSION: \(.*\) -->$/\1/')" || return 1
+   local ui_version="$(grep '<!-- CONSUL_VERSION: .* -->' "$1" | sed 's/<!-- CONSUL_VERSION: \(.*\) -->/\1/' | xargs)" || return 1
```

Comment:
Why was this change necessary?

Reply:
Regex wasn't matching properly anymore due to whitespacing issues. 
Ex:
```
ERROR: UI version mismatch. Expecting: 'v1.5.1-8-g33bfddb90-dev (33bfddb90)' found '  <!-- CONSUL_VERSION: v1.5.1-8-g33bfddb90-dev (33bfddb90) -->'
```

Reply:
Previously the things we had to print into the `index.html` file where reasonably simple.

https://github.com/hashicorp/consul/blob/master/ui-v2/lib/startup/index.js, note the single line CONSUL_VERSION.

In order to make this feature work we have to do slightly more logic and print more things out.

The templating engine that ember uses to produce `index.html` _looks_ like handlebars, but is not. It's like a 'handlebars lite', so you don't have access to all of the template helper functions and I don't think you can easily pass in variables etc.

So we moved as much as we could to use native javascript templates instead, meaning we can pass in variables and make things a lot easier to work with, which included indentation:

https://github.com/hashicorp/consul/blob/1930-mod-context-path/ui-v2/lib/startup/templates/head.html.js

Once we included indentation, we realized the grepping for the CONSUL_VERSION string stopped working because of the `^..$`, so we removed it for a looser regex.